### PR TITLE
Adding a host alias fetch from the kubelet

### PR DESCRIPTION
**Pull request by jiquiame**
### What does this PR do?

Making the host alias logic fetching the hostname used by kubernetes.

### Motivation

So that the metrics from kubernetes_state no longer create duplicated hosts as the host alias from kubernetes is different from the ones that are already fetched.

### CODE REVIEW

File: pkg/util/kubernetes/hostinfo/host_alias.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubelet,kubeapiserver
```
[the remark this answers is not in the document]
Comment (antoinepouille) [reply pkg/util/kubernetes/hostinfo/host_alias.go]: 👍 Thanks for the details about build tags!

File: pkg/util/kubernetes/hostinfo/host_alias.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+)
+
+// GetHostAlias uses the "kubelet" hostname provider to fetch the kubernetes alias
+func GetHostAlias() (string, error) {
+	if getKubeletHostname, found := hostname.ProviderCatalog["kubelet"]; found {
```
[the remark this answers is not in the document]
Comment (antoinepouille) [reply pkg/util/kubernetes/hostinfo/host_alias.go]: Yep, no need to use the ProviderCatalog here. I was worried with other cyclic dependancies, but we're good here

**Review by xvello — APPROVED** [review]
LGTM, let's get croissant to confirm that should not bring any side-effects

**Review by olivielpeau — APPROVED** [review]
👍 LGTM as long as there's a timeout (cumulatively < 1sec) on any network call that `kubelet.HostnameProvider` might make :)

In general adding host aliases is pretty safe as long as they're actually unique to each host

**Pull request merged**